src/docs/rust.mdx

```diff
@@ -0,0 +1,323 @@
+
+By default, fields of structs (including tuple-structs) should be fully private. The only exception to this are:
+
+- Newtypes (1-tuple structs) where direct access to the inner type is desired. This is to annotate semantics of a type where accessing the inner type is required.
+- Plain data types with a very stable signature.
+
+Mixed visibility is never allowed, not even between private and `pub(crate)` or `pub(super)`. Instead, provide accessors:
+
+- `foo()`, `foo_mut()` and `set_foo()` for exposing the values
+- `pub(crate)` or `pub(super)` accessor functions in corner cases
+- A `Default` implementation and builder when incremental construction is required
+
+### Naming
+
+We are strictly following the [Rust Naming Conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html). The entire conventions apply, although here are a few highlighted sections:
+
+- [Avoid redundant prefixes](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#avoid-redundant-prefixes-[rfc-356])
+- [Getter and setter methods](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#getter/setter-methods-[rfc-344]) (`foo` instead of `get_foo`)
+- [Conversions](https://doc.rust-lang.org/1.0.0/style/style/naming/conversions.html) (`as_foo` vs `to_foo` vs `into_foo`)
+- [Iterators](https://doc.rust-lang.org/1.0.0/style/style/naming/iterators.html)
+- [Constructors](https://rust-lang.github.io/api-guidelines/predictability.html#constructors-are-static-inherent-methods-c-ctor) (Structs generally have `fn new(...) -> Self`, which never returns Result)
+
+### Writing Doc Comments
+
+We follow conventions from RFC 505 and RFC 1574 for writing doc comments. See the [full conventions text](https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text). Particularly, this includes:
+
+- A single-line short summary sentence written in American English using third-person.
+- Use the default headers where applicable, but avoid custom sections outside of module-level docs.
+- Link between types and methods where possible, especially within the crate.
+- Write doc tests
+
+### Writing Doc Tests
+
+If possible, prefer writing at least one doctest for the critical path of your methods or structs. Of course, this requires the interface to be public. This should even take precedence over an equal unit test, since it both tests and documents the API.
+
+To format code in doc comments, you can temporarily [configure rustfmt](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#format_code_in_doc_comments) to do so. We might add this to our default configuration at some point:
+
+```rust
+format_code_in_doc_comments = true
+```
+
+### Writing Tests
+
+All test functions should start with `test_` and contain the function name + a simple condition in their name. The test name should be as concise as possible and avoid redundant words (such as "should", "that"). Examples:
```

Comment:
Why exactly is that though? IMO this is redundant and pure noise. Things are marked with `#[test]` anyway, Rust does not use magic prefixes to denote tests.

Reply:
Discussed this with @jan-auer and agree that `test_` is a redundant prefix. Removed this rule and updated the examples.